dev/openmage/nginx-frontend.conf

```diff
@@ -0,0 +1,191 @@
+geo $whitelist_remote_addr {
+    default 0;
+    # CIDR in the list below are not limited
+    10.0.0.0/8 1;
+    172.16.0.0/12 1;
+    192.168.0.0/16 1;
+}
+map $whitelist_remote_addr $limit_key {
+    0     $binary_remote_addr;
+    1     "";
+}
+map $http_x-magentoruncode $runcode {
+    default "${NGINX_MAGE_RUN_CODE}";
+}
+map $http_x-magentoruntype $runtype {
+    default "${NGINX_MAGE_RUN_TYPE}";
+}
+map $request_method $post_limit {   # Setup a special rate limiter for POST requests
+    default         "";
+    POST            $limit_key;
+}
+map $http_x_forwarded_proto $fastcgi_param_https {
+    default '';
+    https 'on';
+}
+
+# Trust X-Forwarded-For header coming from local networks
+set_real_ip_from  10.0.0.0/8;
+set_real_ip_from  172.16.0.0/12;
+set_real_ip_from  192.168.0.0/16;
+real_ip_header    X-Forwarded-For;
+real_ip_recursive on;
+
+limit_req_zone $limit_key   zone=spam:10m      rate=4r/m;
+limit_req_zone $post_limit  zone=post:10m      rate=1r/s;
+limit_req_zone $limit_key   zone=get:10m       rate=3r/s;
+limit_req_zone $limit_key   zone=media:10m     rate=10r/s;
+limit_req_log_level warn;
+
+server {
+    listen 80;
+    server_name _ "";
+
+    if ( $request_method !~ ^(GET|POST|HEAD|OPTIONS)$) { return 405; }
+
+    set $fastcgi_script_filename "";  # Required because it is used in a log format
+    set $fastcgi_script_name2 $fastcgi_script_name; # Allow this to be manipulated
+    set $fastcgi_php_value "";        # Pre/Append to add more PHP FPM config
+
+    client_max_body_size 16K;         # Default max body size - increase using a specific location block
+
+    gzip on;
+    gzip_comp_level 6;
+    gzip_min_length 1100;
+    gzip_buffers 16 8k;
+    gzip_proxied any;
+    gzip_types
+        text/plain
+        text/css
+        text/js
+        text/xml
+        text/javascript
+        application/javascript
+        application/x-javascript
+        application/json
+        application/xml
+        application/xml+rss
+        image/svg+xml;
+    gzip_vary on;
+
+    location / {
+        limit_req zone=get burst=10 nodelay;
+        limit_req zone=post burst=3 nodelay;
+        root /pub/$runcode;                                           # Only files in this directory can be loaded directly
+        set $fastcgi_script_filename /var/www/html/index.php;         # Hard-coded to prevent loading other PHP files
+        try_files $uri @php-fpm;
+
+        # Rate limit for specific endpoints that are commonly spammed
+        location /newsletter/subscriber/new {
+            limit_req zone=spam;
```

Comment:
i usually add also:
```
valid_referers server_names;
if ($invalid_referer) {
    return 403;
}
```

Reply:
The server name is defined as `server_name _ "";` which is a catch-all so I don't think that will work the way this config is designed. I don't disagree with adding it, but wouldn't the form key already be a redundant/stronger mitigation than checking the Referrer header?

Reply:
Indeed, I'm not sure if it works with server_name _ "";. 
Yes, form_key can help mitigate the issue if set in forms. I believe I added this control when the some forms didn't already have a form_key, like the newsletter.
I think now, with an updated version of OpenMage, it's useless unless the checks on form_key are disabled from backend.